app/scripts/lib/local-store.js

```diff
@@ -0,0 +1,25 @@
+// We should not rely on local storage in an extension!
+// We should use this instead!
+// https://developer.mozilla.org/en-US/Add-ons/WebExtensions/API/storage/local
+
+const extension = require('extensionizer')
+const STORAGE_KEY = 'metamask-config'
+
+module.exports = class ExtensionStore {
+  constructor() {
+    this.isSupported = !!(extension.storage.local)
+    if (!this.isSupported) {
+      log.error('Storage local API not available.')
+    }
+  }
+  async get() {
```

Comment:
doesnt need to be marked as `async` when manually returning a promise but maybe this is a good practice anyways